File: start.go
package main

import (
    "fmt"
    "log"
    "os"
    "flag"
    "github.com/PuerkitoBio/goquery"
)


func check(e error) {
    if e != nil {
        log.Fatal(e)
        panic(e)
    }
}

func main() {

  resp, err := goquery.NewDocument("http://gobyexample.com")

  check(err)

  flag.Parse()

  var baseDir string

  cmdArg := flag.Args()

  if len(cmdArg) != 0 {
      baseDir = cmdArg[0]
  } else {
      baseDir = "learn-go-by-example"
  }

  os.Mkdir(baseDir, os.FileMode(0777))

  resp.Find("li > a").Each(func(i int, s *goquery.Selection) {
      a, _ := s.Attr("href")

      fileName := fmt.Sprintf("%02d-%v", i, a)

      filePath := fmt.Sprintf("%v/%v", baseDir, fileName)

      os.Mkdir(filePath, os.FileMode(0777))

      path := fmt.Sprintf("%v/%v.go", filePath, fileName)

      f, err := os.Create(path)

      check(err)

      defer f.Close()
  })
}

File: start.js
var fs = require('fs');
var axios = require('axios');
var cheerio = require('cheerio');
var url = 'https://gobyexample.com';
var baseDir = process.argv[2] || 'learn-go-by-example';

function check(error) {
  if (error) {
    throw error;
  }
}

fs.mkdir(baseDir, function(error) {
  check(error);
  axios.get(url).then(function(response) {
    var $ = cheerio.load(response.data);
    $('li > a').each(function(index, element) {
      let num = index < 10 ? '0' + index : index;
      let filePath = `${num}-${$(element).attr('href')}`;
      let fileName = `${filePath}.go`;
      let partialPath = `${baseDir}/${filePath}`;
      let fullPath = `${baseDir}/${filePath}/${fileName}`;
      fs.mkdir(partialPath, function(error) {
        check(error);
        fs.writeFile(fullPath, '', function(error) {
          check(error);
        });
      });
    });
  }).catch(function(error) {
    check(error);
  });
});
